File: backend/src/main/java/com/healthchain/backend/model/DocumentReference.java
package com.healthchain.backend.model;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class DocumentReference {
    private String resourceType;
    private String id;
    private Reference subject;
    private String date;
    private List<Reference> author;
    private String description;
    private List<Content> content;

    public static class Reference {
        private String reference;
    }

    public static class Content {
        private Attachment attachment;
    }

    public static class Attachment {
        private String contentType;
        private String url;
        private String data;
        private String title;
    }
}

File: network/custom-chaincode/chaincode/lib/model/DocumentReference.js
'use strict';

class DocumentReference {
    constructor(id, subject, date, author, description, content) {
        this.resourceType = 'DocumentReference';
        this.id = id;
        this.subject = subject; //Reference
        this.date = date;
        this.author = author;
        this.description = description;
        this.content = content; //List<Attachment>
    }
}

class Reference {
    constructor(reference) {
        this.reference = reference;
    }
}

class Content {
    constructor(attachment) {
        this.attachment = attachment;
    }
}

class Attachment {
    constructor(contentType, url, data, title) {
        this.contentType = contentType;
        this.url = url;
        this.data = data;
        this.title = title;
    }
}

module.exports = {
    DocumentReference,
    Reference,
    Content,
    Attachment
};